Added BuffTypes arguments.

yasinallahdev opened the pull request:
All calls to AddBuffHudVisual in the Subrepo now use a BuffType argument on creation. This Pull Request should not be merged until LeagueSandbox/GameServer#711 is merged.

Commit:
Added BuffTypes arguments.

All calls to AddBuffHudVisual in the Subrepo now use a BuffType argument on creation.

Commit:
Updating Highlander Script

I'd forgotten to add a BuffType argument to Master Yi's Highlander. That's a big 'oops' from me.

MatthewFrench commented on Champions/Global/SummonerRevive.cs:
```diff
@@ -27,10 +28,9 @@ public void OnFinishCasting(Champion owner, Spell spell, AttackableUnit target)
+            var visualBuff = ApiFunctionManager.AddBuffHudVisual("SummonerReviveSpeedBoost", 12.0f, 1, BuffType.COMBAT_ENCHANCER, owner, 12.0f);
             ApiFunctionManager.CreateTimer(12.0f, () =>
             {
-                ApiFunctionManager.RemoveBuffHudVisual(visualBuff);
```
Is there a downside to not removing the buff hud visual?

yasinallahdev replied:
The final argument in AddBuffHudVisual is a Timer Argument, which will cause the buff HUD visual to be automatically be removed after the set amount of time. This means that the CreateTimer() calls are (in several cases) unneeded extra code.

MatthewFrench commented on Champions/MasterYi/R.cs:
```diff
@@ -26,11 +27,11 @@ public void OnStartCasting(Champion owner, Spell spell, AttackableUnit target)
             ApiFunctionManager.CreateTimer(10.0f, () =>
             {
                 ApiFunctionManager.RemoveParticle(p);
-                ApiFunctionManager.RemoveBuffHudVisual(visualBuff);
```
Is there a downside to not removing the buff hud visual?

yasinallahdev replied:
See above comment regarding the same question in regards to SummonerRevive. I should probably use AddBuffHudVisual()'s Timer argument in all applicable cases for completion, though.

MatthewFrench reviewed (commented):
Just a question about a few missing hud buff removals. Other than that looks good.

Commit:
Updated Most Scripts to auto-remove buffs.

AddBuffHudVisual contains a removeAfter float argument for auto-removal of the buff, and AddBuffGameScript has a similar argument for BuffGameScripts. These arguments are now in use. Most BuffHudVusual calls are now moved to a BuffGameScript with these changes.
